### PR TITLE
Fixing readout of backslashes (#6301)

### LINKED ISSUE #6301: [bug] virtualenv generator strip '\' on windows when upgrading to conan 1.21.0
Hi,

After updating to 1.21.0, our CI and developer have been experiencing issues on script and tool that relied on conan generated activate.sh/activate_run.sh on windows. It looks like that it may be caused by some recent change on those generator.

### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10
  * Compiler+version:
  * Conan version: 1.21.0
  * Python version: 2.7.12

### Steps to reproduce (Include if Applicable)

Here's the small reproduce step I could come up with

```
conan install cmake_installer/3.15.4@conan/stable -g virtualenv -if build
source build/activate.sh
echo $CMAKE_ROOT
```
### Logs (Executed commands with output) (Include/Attach if Applicable)

With conan 1.21.0
```
$ echo $CMAKE_ROOT
C:Usersjbounthong.conandatacmake_installer3.15.4conanstablepackage60b04a80ac4dd5c23f1acffacc679e9902e636e5
```

With conan 1.20.5
```
$ echo $CMAKE_ROOT
C:\Users\jbounthong\.conan\data\cmake_installer\3.15.4\conan\stable\package\60b04a80ac4dd5c23f1acffacc679e9902e636e5
```

### Cause & Impact

The regression seems to be cause by recent change on virtualenv, virtualrunenv generator, and the introduction of `environment.sh.env`. When it's being sourced by the `activate.sh` the windows `\` directory separators are being stripped down.

We are being impacted by it since we rely on activate_run.sh to set the PATH variable so that built `.exe` execution can lookup dependent `.dll` provided by conan.

### PR DESCRIPTION
**Pull request by kucerad**
Changelog: Bugfix: Fixing readout of backslashes (#6301)
Docs: omit

closes #6301 
closes #6323

### CODE REVIEW

**Comment by jgsogo** [comment]
Thanks a lot! I've merged the test had prepared for this change. Now it's time to review the PR (mostly my test because it is a bit weird) and it will be merged into the next release v1.21.1

Thanks a lot!

File: conans/test/functional/generators/virtualenv_windows_bash_test.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+        cls.cache_folder = os.path.dirname(t.cache_folder)
+
+    def _run_environment(self):
+        stdout, _ = subprocess.Popen(["where", "conan"], stdout=subprocess.PIPE).communicate()
+        conan_path = decode_text(stdout).splitlines()[0]  # Get the first one (Windows returns all found)
+
+        # All the commands are listed in a sh file:
+        commands_file = os.path.join(self.test_folder, 'commands.sh')
+        with open(commands_file, 'w+') as f:
```
Comment (memsharded) [inline conans/test/functional/generators/virtualenv_windows_bash_test.py]: Same here, too many ``f.write()`` you can probably have a clean template that you fill with the folders, and then ``save(commands_file, content)``.

This is a matter of taste I am fine with it as-is too if you want to leave it.

File: conans/test/functional/generators/virtualenv_windows_bash_test.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    def setUpClass(cls):
+        cache_folder = os.path.join(temp_folder(path_with_spaces=False), ".conan")
+        t = TestClient(cache_folder=cache_folder)
+        t.save({"conanfile.py": cls.conanfile})
+        t.run("create . name/version@")
+
+        cls.cache_folder = os.path.dirname(t.cache_folder)
+
+    def _run_environment(self):
+        stdout, _ = subprocess.Popen(["where", "conan"], stdout=subprocess.PIPE).communicate()
```
Comment (memsharded) [inline conans/test/functional/generators/virtualenv_windows_bash_test.py]: I am not sure why Conan needs to be in the environment script. I thought that just managing env-vars and paths, would be enough. This seems to make the test very complicated, but I don't see the need.

File: conans/test/functional/generators/virtualenv_windows_bash_test.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        env_before = _load_env_file(os.path.join(self.test_folder, "env_before.txt"))
+        env_after = _load_env_file(os.path.join(self.test_folder, "env_after.txt"))
+        self.assertDictEqual(env_before, env_after)  # Environment restored correctly
+
+        return stdout, _load_env_file(os.path.join(self.test_folder, "env_activated.txt"))
+
+    def test_basic_variable(self):
+        self.test_folder = temp_folder(path_with_spaces=False)
+        _, environment = self._run_environment()
```
Comment (memsharded) [inline conans/test/functional/generators/virtualenv_windows_bash_test.py]: I am a bit confused. These 3 tests looks that they are running exactly the same environment execution, and they just differ in the checks. In that case, I really think that it should be just 1 test, with all the necessary checks, and also maybe the setUpClass could be simplified.

File: conans/test/functional/generators/virtualenv_windows_bash_test.py [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+        commands_file = os.path.join(self.test_folder, 'commands.sh')
+        with open(commands_file, 'w+') as f:
+            # Dirty environment
+            f.write("export USER_VAR=existing_value\n")
+            f.write("export ANOTHER=existing_value\n")
+            f.write("export WHATEVER=existing_value\n")
+            f.write("export WHATEVER2=existing_value\n")
+            f.write("export CFLAGS=existing_value\n")
+
+            f.write("export PATH={}:$PATH\n".format(os.path.dirname(conan_path).replace('\\', '/').replace('C:', '/c')))
```
Comment (memsharded) [inline conans/test/functional/generators/virtualenv_windows_bash_test.py]: try to make it work with C:/

File: conans/test/functional/generators/virtualenv_windows_bash_test.py (hunk 2 above)
Comment (jgsogo) [reply conans/test/functional/generators/virtualenv_windows_bash_test.py]: Just for the reference, if anyone reads this comment in the future, go to https://github.com/conan-io/conan/issues/6328

File: conans/test/functional/generators/virtualenv_windows_bash_test.py (hunk 4 above)
Comment (jgsogo) [reply conans/test/functional/generators/virtualenv_windows_bash_test.py]: It doesn't work 😞

**Pull request merged**